cli/src/add_url.rs

```diff
@@ -0,0 +1,18 @@
+// File: add_url.rs
+use db::PgPool;
+use color_eyre::Result;
+
+use crate::check_auth_status;
+
+pub async fn add_url(pool: &PgPool, url: &str, allow_existing: bool) -> Result<()> {
+
+    check_auth_status(pool).await?;
+
+    let result = db::add_url(pool, url, &allow_existing).await;
+    match result {
+        Ok(message) if message.contains("re-adding is allowed") => println!("{message}"),
+        Ok(message) => println!("URL added successfully with ID: {message}"),
```

Comment:
Ahh! Here might be a fun case where we want to return something different than just a `String` message so that we can tell these two cases apart!

The easiest answer is maybe we want to return a new Enum instead! We can make a new Enum, maybe called `AddUrlOutcome` (for lack of a more original name lol) with two variants, one for when we created a new URL and one for where an existing one existed!

```
enum AddUrlOutcome {
  Created(String),
  Existing(String),
}
```

And then you can match on the enum more specifically instead of having do a `message.contains` check on the returned message!